### ISSUE TITLE
Passing an enum into a string template is not supported

### ISSUE THREAD

**Issue by matthova** [fabien0102/ts-to-zod#64]
## Bug description
Passing an enum into a string template causes zod generation to fail.

### Input

```ts
export enum Foo {
  Bar = 'Bar',
  Baz = 'Baz',
}

export interface Party {
  Time: `${Foo}`;
}
```

### Expected output

```ts
export const fooSchema = z.nativeEnum(Foo);
export const partySchema = z.object({
  Time: z.union([z.literal('Bar'), z.literal('Baz')]),
});
```

### Actual output
 »   Warning: 'TemplateLiteralType' is not supported, fallback into 'z.any()'
✖ Validating generated types
 ›   Error: 'partySchema' is not compatible with 'Party':
 ›   Argument of type '{ Time?: any; }' is not assignable to parameter of type 'Party'.
 ›     Property 'Time' is optional in type '{ Time?: any; }' but required in type 'Party'.
error Command failed with exit code 2.

**Versions**

- Typescript: `v4.5.2`
- Zod: `v3.11.6`

**Comment by fabien0102** [comment]
Whooo, this is a tricky one 😅 Sounds like a nice challenge! I putting this into my challenges todolist 😁 

Thanks for the report!

**Comment by tvillaren** [comment]
Fixed in [3.8.0](https://github.com/fabien0102/ts-to-zod/releases/tag/v3.8.0)
